src/GH.hs

```diff
@@ -44,20 +52,33 @@ data URLParts = URLParts
   , tag :: Text
   }
 
+-- | Parse owner-repo-branch triplet out of URL
+-- We accept URLs pointing to uploaded release assets
+-- that are usually obtained with fetchurl, as well
+-- as the generated archives that fetchFromGitHub downloads.
+--
+-- Examples:
+--
+-- >>> runExceptT (parseURL "https://github.com/blueman-project/blueman/releases/download/2.0.7/blueman-2.0.7.tar.xz")
```

Comment:
I am not really sure how to test this. We probably have to split the function and operate on `Maybe`.